services/spar/test/Test/Spar/Scim/UserSpec.hs

```diff
@@ -0,0 +1,189 @@
+module Test.Spar.Scim.UserSpec where
+
+import Arbitrary ()
+import Brig.Types.Intra
+import Brig.Types.User
+import Control.Monad.Except (runExceptT)
+import Data.Handle (parseHandle)
+import Data.Id
+import qualified Data.Json.Util
+import Imports
+import Polysemy
+import Polysemy.TinyLog
+import Spar.Scim.User (deleteScimUser)
+import Spar.Sem.BrigAccess
+import Spar.Sem.IdPConfigStore
+import Spar.Sem.IdPConfigStore.Mem (TypedState, idPToMem)
+import Spar.Sem.SAMLUserStore
+import Spar.Sem.SAMLUserStore.Mem (UserRefOrd, samlUserStoreToMem)
+import qualified Spar.Sem.ScimExternalIdStore as ScimExternalIdStore
+import Spar.Sem.ScimExternalIdStore.Mem (scimExternalIdStoreToMem)
+import Spar.Sem.ScimUserTimesStore
+import Spar.Sem.ScimUserTimesStore.Mem (scimUserTimesStoreToMem)
+import System.Logger (Msg)
+import Test.Hspec
+import Test.QuickCheck
+import Web.Scim.Schema.Error
+import Wire.API.User
+import qualified Wire.API.User.Identity
+import Wire.API.User.Scim
+import Wire.Sem.Logger.TinyLog (discardTinyLogs)
+
+spec :: Spec
+spec = describe "deleteScimUser" $ do
```

Comment:
This is a nice case study on how to write tests based on polysemy, and it's perfectly mergeable and useful.

In the long run, I wonder how we can make this more convenient.  If we want to replace arbitrary interpreters out of a composition of 5-15 of them, things will become awkward.  Organising interpreters in trees (ie., building composite effects from more basic ones) would probably help?  Or what about a record that contains all the default interpreters and that is passed declared somewhere in the test suite and a function that builds a composite interpreter from all the record fields; and then we can replace individual fields in the record using lenses before passing it to that function?

@isovector iff you are curious, feel free to form an opinion while you're waiting for the new contract.  (no pressure :))

Reply:
Good question. Interpreters are a DAG that we're unfortunate enough to need to encode as a list. The hard part here is tracking effect dependencies, but some sort of staged "compilation" could be work. I'll put some thought into it.